backend/src/utils/transactionUtils.js

```
@@ -0,0 +1,167 @@
+        'P2025', // Record not found
+        'P2027', // Multiple records found
+        'P2034', // Transaction failed
+    ];
+    
+    // Check if it's a Prisma error with a non-retryable code
+    if (error.code && nonRetryableErrors.includes(error.code)) {
+        return true;
+    }
+    
+    // Check for connection errors that shouldn't be retried
+    if (error.message && error.message.includes('Connection')) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)